src/net_processing.cpp

```diff
@@ -3841,6 +3839,10 @@ bool PeerLogicValidation::ProcessMessages(CNode* pfrom, std::atomic<bool>& inter
     }
     CNetMessage& msg(msgs.front());
 
+    if (gArgs.IsArgSet("-logmessages")) {
```

Comment:
here and in `net.cpp::2834`, this may be somewhat more robust:

```cpp
if (gArgs.GetBoolArg("-logmessages", false)) {
```

see https://github.com/bitcoin/bitcoin/pull/19089/files#diff-321303fddcf725df060981d626a05df9R619 and tests at https://github.com/bitcoin/bitcoin/pull/19089/files#diff-ff77e31823822bc143e3113aaf85d4b9R64

Reply:
thanks, fixed